lib/krane/kubernetes_resource/custom_resource_definition.rb

```diff
@@ -47,7 +47,7 @@ def kind
     end
 
     def name
-      @definition.dig("metadata", "name")
+      @definition.dig("metadata", "name") || @definition.dig("metadata", "generateName")
```

Comment:
Hm, I wonder why we have this as a method in this one class only, and why the method isn't just an accessor for `@name`. It's not like this name source is unusual. Can you try removing this? The superclass already has `attr_reader :name`

Reply:
I tried removing it to see what would happen. There are other tests failing so I'm actually not sure. It didn't cause any _extra_ tests to fail so maybe you're right, it can 🔥. I'll leave it for now and come back to it when I'm able to get the rest of my tests passing.